cdap-api-common/src/main/java/co/cask/cdap/api/data/format/StructuredRecord.java

```diff
@@ -71,6 +81,160 @@ public Schema getSchema() {
     return (T) fields.get(fieldName);
   }
 
+  /**
+   * Get {@link LocalDate} from field of logical type date.
+   *
+   * @param fieldName date field to get.
+   * @return value of the field as an object of {@link LocalDate}, returns null if given field has null value
+   * @throws IllegalArgumentException if the field name is not present in schema or the field provided
+   *                                  is not of {@link Schema.LogicalType#DATE} type.
+   */
+  public LocalDate getDate(String fieldName) throws IllegalArgumentException {
+    validateAndGetLogicalType(fieldName, schema.getField(fieldName),
+                              new HashSet<>(Collections.singletonList(Schema.LogicalType.DATE.name())));
+    return this.fields.get(fieldName) == null ? null : LocalDate.ofEpochDay(((Integer) fields.get(fieldName))
+                                                                              .longValue());
+  }
+
+  /**
+   * Get {@link LocalTime} from field of logical type time.
+   *
+   * @param fieldName time field to get.
+   * @return value of the field as an object of {@link LocalTime}, returns null if given field has null value
+   * @throws IllegalArgumentException if the field name is not present in schema or the field provided
+   *                                  is not of {@link Schema.LogicalType#TIME_MILLIS} or
+   *                                  {@link Schema.LogicalType#TIME_MICROS} type.
+   */
+  public LocalTime getTime(String fieldName) {
+    Schema.LogicalType logicalType =  validateAndGetLogicalType(fieldName, schema.getField(fieldName),
+                                                                new HashSet<>(
+      Arrays.asList(Schema.LogicalType.TIME_MILLIS.name(), Schema.LogicalType.TIME_MICROS.name())));
+
+    if (fields.get(fieldName) == null) {
+      return null;
+    }
+
+    if (logicalType == Schema.LogicalType.TIME_MILLIS) {
+      return LocalTime.ofNanoOfDay(TimeUnit.MILLISECONDS.toNanos(((Integer) fields.get(fieldName))));
```

Comment:
Remember the value of `fields.get()` as `Object` instead of calling it multiple times.